src/components/dashboard/DashboardPage.jsx

```diff
@@ -82,12 +91,34 @@ const DashboardPage = () => {
           <Tab eventKey="courses" title="Courses">
             {CoursesTabComponent}
           </Tab>
-          <Tab eventKey="programs" title="Programs">
-            <ProgramListingPage />
+          <Tab eventKey="programs" title="Programs" disabled={learnerProgramsListData.length === 0}>
+            {learnerProgramsListData > 0
+            && (
+              <ProgramListingPage
+                canOnlyViewHighlightSets={canOnlyViewHighlightSets}
+                programData={
+                  {
+                    data: learnerProgramsListData,
+                    error: programsFetchError,
+                  }
+                }
+              />
+            )}
           </Tab>
           {features.FEATURE_ENABLE_PATHWAY_PROGRESS && (
```

Comment:
nit: unrelated to your specific changes, but this conditional rendering of a `Tab` isn't supported for the `Tabs` component and is likely throwing console.warn in the devtools console about it. Rather than conditionally rendering `Tab` inline here, you might create a `tabs` array to pass as the children to `Tabs` instead.